resolver/nns.go

```diff
@@ -0,0 +1,125 @@
+package resolver
+
+import (
+	"context"
+	"fmt"
+
+	nns "github.com/nspcc-dev/neo-go/examples/nft-nd-nns"
+	"github.com/nspcc-dev/neo-go/pkg/rpc/client"
+	"github.com/nspcc-dev/neo-go/pkg/rpc/response/result"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+)
+
+type NNSResolver interface {
+	// ResolveContainerName get record for domain name+'.container' from NNS contract.
+	ResolveContainerName(name string) (string, error)
+}
+
+type nnsResolver struct {
+	rpc     *client.Client
+	nnsHash util.Uint160
+}
+
+const (
+	resolve         = "resolve"
+	containerDomain = ".container"
+)
+
+// NewNNSResolver creates resolver that can get records from NNS contract.
+func NewNNSResolver(ctx context.Context, address string) (NNSResolver, error) {
+	rpc, err := client.New(ctx, address, client.Options{})
+	if err != nil {
+		return nil, err
+	}
+	if err = rpc.Init(); err != nil {
+		return nil, err
+	}
+
+	nnsContract, err := rpc.GetContractStateByID(1)
+	if err != nil {
+		return nil, err
+	}
+
+	return &nnsResolver{
+		rpc:     rpc,
+		nnsHash: nnsContract.Hash,
+	}, nil
+}
+
+func (n *nnsResolver) ResolveContainerName(name string) (string, error) {
```

Comment:
The same comments as for system wide resolver. `cid.ID` and try all available TXT record until you find record with CID.